### PR TITLE
[webgpu] Add support for gatherND

### PR DESCRIPTION
**Pull request by haoyunfeix**
FEATURE

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5057)

### CODE REVIEW

File: tfjs-backend-webgpu/src/kernels/gather_nd_webgpu.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        ${stridesType} strides = ${stridesType}(${this.strides});
+         void main() {
+          int currentIndex = int(gl_GlobalInvocationID.x);
+          ${dtype} coords = getOutputCoords();
+          int flattenIndex = 0;
+          for (int j = 0; j < ${this.sliceDim}; j++) {
+            int index = int(round(getIndices(coords[0], j)));
+            flattenIndex += index * ${strideString};
+          }
+          if (currentIndex < ${this.size}) {
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/gather_nd_webgpu.ts:62]: You can directly use `if (currentIndex < size)`

File: tfjs-backend-webgpu/src/kernels/gather_nd_webgpu.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  constructor(sliceDim: number, strides: number[], shape: number[]) {
+    this.outputShape = shape;
+    this.dispatchLayout = flatDispatchLayout(this.outputShape);
+    this.dispatch = computeDispatch(
+        this.dispatchLayout, this.outputShape, this.workGroupSize);
+    this.shaderKey = `gathernd_${sliceDim}`;
+    this.size = util.sizeFromShape(this.outputShape);
+    this.sliceDim = sliceDim;
+    this.uniforms =
+        `int sliceDim; ${getCoordsDataType(strides.length)} strides;`;
```
Comment (qjia7) [inline tfjs-backend-webgpu/src/kernels/gather_nd_webgpu.ts:44]: Is `sliceDim` equal to `strides.length`? If yes, you can remove the parameter `strides: number[]` .

Comment (qjia7) [reply tfjs-backend-webgpu/src/kernels/gather_nd_webgpu.ts:44]: It seems that it is. See https://github.com/tensorflow/tfjs/blob/master/tfjs-core/src/ops/gather_nd_util.ts#L82. Then you can write it like below:
```
constructor(sliceDim: number, shape: number[])
...
this.uniforms =
        `int sliceDim; ${getCoordsDataType(sliceDim)} strides;`;

**Review by qjia7 — APPROVED** [review]
LGTM
@lina128 @jinjingforever take a look, thanks.

**Review by jinjingforever — APPROVED** [review]

**Pull request merged**